src/main/java/com/epam/ta/reportportal/demo_data/DemoLogsService.java

```diff
@@ -27,90 +27,137 @@
-			int t = random.nextInt(30);
-			List<Log> logs = IntStream.range(1, t + 1).mapToObj(it -> {
-				Log log = new Log();
-				log.setLevel(logLevel());
-				log.setLogTime(new Date());
-				log.setTestItemRef(itemId);
-				log.setLogMsg(logMessages.get(random.nextInt(logMessages.size())));
-				return log;
-			}).collect(toList());
-			if (FAILED.name().equals(status)) {
-				String file = dataStorage.saveData(new BinaryData(IMAGE_PNG_VALUE, img.contentLength(), img.getInputStream()), "file");
-                logs.addAll(errorLogs.stream().map(msg -> {
-                    Log log = new Log();
-                    log.setLevel(ERROR);
-                    log.setLogTime(new Date());
-                    log.setTestItemRef(itemId);
-                    log.setLogMsg(msg);
-                    final BinaryContent binaryContent = new BinaryContent(file, file, IMAGE_PNG_VALUE);
-                    log.setBinaryContent(binaryContent);
-                    return log;
-                }).collect(toList()));
-			}
-			return logRepository.save(logs);
-		} catch (IOException e) {
-			throw new ReportPortalException("Unable to generate demo logs", e);
-		}
-	}
+    @Autowired
+    DemoLogsService(DataStorage dataStorage, LogRepository logRepository) {
+        this.dataStorage = dataStorage;
+        this.logRepository = logRepository;
+        this.random = new SplittableRandom();
+    }
 
-	private LogLevel logLevel() {
-		int i = random.nextInt(50);
-		if (i < 10) {
-			return DEBUG;
-		} else if (i < 20) {
-			return WARN;
-		} else if (i < 30) {
-			return TRACE;
-		} else {
-			return INFO;
-		}
-	}
+    List<Log> generateDemoLogs(String itemId, String status) {
+        int t = random.nextInt(30) + 5;
+        List<Log> logs = IntStream.range(1, t + 1).mapToObj(it -> {
+            Log log = new Log();
+            log.setLevel(logLevel());
+            log.setLogTime(new Date());
+            if (random.nextInt(t) <= t / 8) {
+                log.setBinaryContent(attachBinaryContent());
+            }
+            log.setTestItemRef(itemId);
+            log.setLogMsg(ContentUtils.getLogMessage());
+            return log;
+        }).collect(toList());
+        if (FAILED.name().equals(status)) {
+            List<String> errors = ContentUtils.getErrorLogs(2);
+            logs.addAll(errors.stream().map(msg -> {
+                Log log = new Log();
+                log.setLevel(ERROR);
+                log.setLogTime(new Date());
+                log.setTestItemRef(itemId);
+                log.setLogMsg(msg);
+                BinaryContent binaryContent = attachBinaryContent();
+                log.setBinaryContent(binaryContent);
+                return log;
+            }).collect(toList()));
+        }
+        return logRepository.save(logs);
+    }
+
+    private BinaryContent attachBinaryContent() {
+        ClassPathResource resource;
+        String contentType;
+        switch (random.nextInt(20)) {
```

Comment:
assign to enum, then generate random like random.nextInt(myenum.values().length)
return should have method that returns Resource